app/controllers/box_design_controller.rb

```diff
@@ -0,0 +1,24 @@
+class BoxDesignController < ApplicationController
+  before_action :authenticate_user!
+
+  def claim
+    @box = box_claim_scope.find(params[:box_id])
+    @box.designed_by = current_user
+    @box.save!
+  end
+
+  def mark_as_designed
+    @box = box_claim_scope.find_by(designed_by: current_user)
+    @box.update(box_design_params)
```

Comment:
This line totally does not work, and I am going to work on fixing it in my next PR